src/server/tablet_snap.rs

```diff
@@ -1,82 +1,157 @@
 // Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
 
+//! This file contains the implementation of sending and receiving tablet
+//! snapshot.
+//!
+//! Different from v1, tablet snapshot always tries to use cache to speed up
+//! transfering speed. The protocol is quite simple:
+//!
+//! sender                            receiver
+//! send snapshot meta    ---->     receive snapshot meta
+//! extra snapshot preview          collect cache meta
+//! send all preview      ---->     receive preview and clean up miss cache
+//! files receive files list      <-----  send missing file list
+//! send missing files    ---->     receive missing files
```

Comment:
Is there other extra way to ensure the data integrity in receiver side?
For example, if the sender does not send all files due to some bug, how can receiver detect it.
Or what if the receiver's missing file list is wrong at first place.